javatests/build/bazel/tests/integration/BazelBaseTestCaseTest.java

```diff
@@ -14,19 +14,201 @@
 
 package build.bazel.tests.integration;
 
+import org.hamcrest.Description;
+import org.hamcrest.SelfDescribing;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+import static org.hamcrest.core.Is.is;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
 import static com.google.common.truth.Truth.assertThat;
 import static org.junit.Assert.assertEquals;
 
-import org.junit.Test;
-
-/** {@link BazelBaseTestCase}Test */
+/**
+ * {@link BazelBaseTestCase}Test
+ */
+//suppress since same parameter value is ok for tests readability, tests should encapsulate and not hide
+@SuppressWarnings("SameParameterValue")
 public final class BazelBaseTestCaseTest extends BazelBaseTestCase {
 
-  @Test
-  public void testVersion() throws Exception {
-    Command cmd = bazel("info", "release");
-    assertEquals(0, cmd.run());
-    assertThat(cmd.getOutputLines())
-        .contains("release " + System.getProperty("bazel.version"));
-  }
+    private static final String WORKSPACE_NAME = "workspace(name = \"build_bazel_integration_test\")\n";
+
+    @Test
+    public void testVersion() throws Exception {
+        Command cmd = bazel("info", "release");
+        assertEquals(0, cmd.run());
+        assertThat(cmd.getOutputLines())
+                .contains("release " + System.getProperty("bazel.version"));
+    }
+
+    @Test
+    public void testTestSuiteExists() throws Exception {
+        loadIntegrationTestRuleIntoWorkspace();
+        setupPassingTest("IntegrationTestSuiteTest");
+
+        Command cmd = bazel("test", "//:IntegrationTestSuiteTest");
+        final int exitCode = cmd.run();
+
+        org.hamcrest.MatcherAssert.assertThat(exitCode, is(successfulExitCode(cmd, workspace)));
+    }
+
+    private TypeSafeDiagnosingMatcher<Integer> successfulExitCode(final Command cmd, final File workspace) {
```

Comment:
Should that be a utility moved in the base test class?

Reply:
So this is what I wrote above. Sorry it's a wall of text but it's important.
This one specifically is in the "maybe" section, I can move it up now but I'm not yet certain it's generic enough. If you think it is then maybe that's the second vote :)

Reply:
I am fine with not moving it now if you prefer (and do it in a follow-up change) but it is definitely a nice utilty you want to use for other stuff :)

Reply:
I'll add it to the following PRs. Basically I have around 6-7 small follow up PRs I want to submit